SQL: Panic when agg function used inside CASE WHEN

```
$ super -version
Version: 5960c78ac

$ super -c "SELECT CASE WHEN AVG(2) BETWEEN 1 AND 3 THEN 1 END;"
panic: (*sem.AggRef) 0xc0004d9b18

goroutine 1 [running]:
github.com/brimdata/super/compiler/semantic/sem.CopyExpr({0xad10d40?, 0xc0004d9b18})
	/Users/phil/work/super/compiler/semantic/sem/expr.go:425 +0xfaa
github.com/brimdata/super/compiler/semantic.(*translator).expr(0xc00050d700, {0xad0fa30?, 0xc0005487e0}, {0xad04d00, 0xc0004ee600})
	/Users/phil/work/super/compiler/semantic/expr.go:61 +0x2ae5
github.com/brimdata/super/compiler/semantic.(*translator).semCaseExpr-range1(...)
	/Users/phil/work/super/compiler/semantic/expr.go:668
github.com/brimdata/super/compiler/semantic.(*translator).semCaseExpr.Backward[...].func1(...)
	/usr/local/opt/go/libexec/src/slices/iter.go:29
github.com/brimdata/super/compiler/semantic.(*translator).semCaseExpr(0xc00050d700, 0xc000548a80, {0xad04d00, 0xc0004ee600})
	/Users/phil/work/super/compiler/semantic/expr.go:667 +0x2ef
github.com/brimdata/super/compiler/semantic.(*translator).expr(0xc00050d700, {0xad0f6a0?, 0xc000548a80}, {0xad04d00, 0xc0004ee600})
	/Users/phil/work/super/compiler/semantic/expr.go:72 +0x1d45
github.com/brimdata/super/compiler/semantic.(*translator).projection(0xc00050d700, {0xc00038d4a0, 0x1, 0x1?}, {0xad04d00, 0xc0004ee600})
	/Users/phil/work/super/compiler/semantic/sql.go:208 +0x68
github.com/brimdata/super/compiler/semantic.(*translator).sqlSelect(0xc00050d700, 0xc000546fa0, {0x0, 0x0, 0x0?}, {0x0?, 0xab2dd40?, 0x0?}, {0x0, 0x0})
	/Users/phil/work/super/compiler/semantic/sql.go:86 +0x785
github.com/brimdata/super/compiler/semantic.(*translator).sqlQueryBody(0x0?, {0xad0ee90?, 0xc000546fa0?}, {0x0?, 0x0?, 0x0?}, {0x0?, 0x0?, 0x0?}, {0x0, ...})
	/Users/phil/work/super/compiler/semantic/sql.go:436 +0x2d8
github.com/brimdata/super/compiler/semantic.(*translator).sqlQueryBody(0xc00050d700, {0xad0ea10?, 0xc00038d400}, {0x0?, 0x0?, 0x0?}, {0x0, 0x0, 0x0}, {0xad04c60, ...})
	/Users/phil/work/super/compiler/semantic/sql.go:448 +0xf85
github.com/brimdata/super/compiler/semantic.(*translator).semOp(0xc00050d700, {0xad0f490?, 0xc0004e7aa0}, {0x0, 0x0, 0x0}, {0xad04c60, 0xbeeaa40})
	/Users/phil/work/super/compiler/semantic/op.go:572 +0x395e
github.com/brimdata/super/compiler/semantic.(*translator).seq(0xc00050d700, {0xc0004e5000?, 0xc0005df8c8?, 0xc00041bab0?}, {0xad04c60?, 0xbeeaa40?})
	/Users/phil/work/super/compiler/semantic/op.go:41 +0x134
github.com/brimdata/super/compiler/semantic.Analyze({0xad17800?, 0xc0004ee300?}, 0xc0004e1b20, 0x8d37b25?, 0x0)
	/Users/phil/work/super/compiler/semantic/analyzer.go:27 +0x276
github.com/brimdata/super/compiler.Analyze(...)
	/Users/phil/work/super/compiler/package.go:24
github.com/brimdata/super/compiler.CompileWithAST(0xc0004ee300, 0x8e66c45?, 0xc00041bab0, 0x1, 0xc, {0x0, 0x0, 0x0})
	/Users/phil/work/super/compiler/package.go:65 +0x65
github.com/brimdata/super/compiler.(*compiler).NewQuery(0xad178a8?, 0xc00038d1d0?, 0xc0005df9f0?, {0x0?, 0x90?, 0xac91bc0?}, 0x8d93500?)
	/Users/phil/work/super/compiler/compiler.go:42 +0x45
github.com/brimdata/super/runtime.CompileQuery({0xad178a8?, 0xc00038d1d0?}, 0xc00054e630, {0xad04440, 0xc00051c060}, 0xc0004e1b20, {0x0, 0x0, 0x0})
	/Users/phil/work/super/runtime/compiler.go:34 +0x10f
github.com/brimdata/super/cmd/super/root.(*Command).Run(0xc0003866c8, {0xc00003e8f0, 0x0, 0x0})
	/Users/phil/work/super/cmd/super/root/command.go:152 +0x54b
github.com/brimdata/super/pkg/charm.path.run({0xc00008aa70, 0x1, 0x1}, {0xc00003e8f0, 0x0, 0x0?})
	/Users/phil/work/super/pkg/charm/path.go:11 +0x7b
github.com/brimdata/super/pkg/charm.(*Spec).Exec(0xbeaf320, {0xc00003e8e0, 0x2, 0x2})
	/Users/phil/work/super/pkg/charm/charm.go:74 +0x1fa
main.main()
	/Users/phil/work/super/cmd/super/main.go:39 +0x5b
```

## Details

Repro is with super version 5960c78.

The repro query above is a simplification of some sqllogictest queries, of which 199 started failing in the [most recent overnight CI run](https://github.com/brimdata/sqllogic-ztests/actions/runs/21164204959).

A binary search reveals that this panic started happening when the changes from #6531 merged.

--- a/compiler/semantic/sem/expr.go
+++ b/compiler/semantic/sem/expr.go
@@ -280,6 +280,11 @@ func CopyExpr(e Expr) Expr {
 			Expr:     CopyExpr(e.Expr),
 			Filter:   CopyExpr(e.Filter),
 		}
+	case *AggRef:
+		return &AggRef{
+			Node:  e.Node,
+			Index: e.Index,
+		}
 	case *ArrayExpr:
 		return &ArrayExpr{
 			Node:  e.Node,

--- a/compiler/ztests/sql/case-with-agg.yaml
+++ b/compiler/ztests/sql/case-with-agg.yaml
@@ -1,0 +1,4 @@
+spq: SELECT CASE WHEN AVG(2) BETWEEN 1 AND 3 THEN 1 END AS out
+
+output: |
+  {out:1}

